x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/ShardChangesIT.java

```diff
@@ -681,6 +691,57 @@ public void testUnfollowIndex() throws Exception {
         assertThat(client().prepareSearch("index2").get().getHits().getTotalHits(), equalTo(2L));
     }
 
+    public void testFailOverOnFollower() throws Exception {
```

Comment:
Double checking - this test fails when you comment out the special retry, right? can you show what the failures look like?

Reply:
I see - this is very subtle and relies on the assertions in the translogs. I rather have it explicit in this test (and hopefully it will be caught by the test most of the time and not with the vague translog assertion). Do you think that my request to check the content of docs rather than just the doc  count will help here?

Reply:
> Do you think that my request to check the content of docs rather than just the doc count will help here?

I think yes. I added "assertSameDocIdsOnShards" to the teardown of this suite. I'll disable the translog assertion and re-run this test.

Reply:
@bleskes I've re-run the test without the translog assertion and the fix. This test was caught by assertConsistentHistoryBetweenTranslogAndLuceneIndex and 
assertSameDocIdsOnShards.

- assertConsistentHistoryBetweenTranslogAndLuceneIndex
```

java.lang.AssertionError: Lucene op [Index{id='281', type='doc', seqNo=280, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}] 
                        translog op [Index{id='281', type='doc', seqNo=280, primaryTerm=2, version=1, autoGeneratedIdTimestamp=-1}]
Expected: <2L>
     but: was <1L>

```

- assertSameDocIdsOnShards
```
java.lang.AssertionError: out of sync shards: primary=[[follower-index][0], node[zuGiLHlfSYaIXn-Pee2xkw], [P], s[STARTED], a[id=p6bpRxC5QpaSXMHXoMNJhw]] num_docs_on_primary=[323] vs replica=[[follower-index][0], node[simUh2x3S9-yQkbi3FH6qA], [R], s[STARTED], a[id=ytqfuUsbSvO0gqHRp1mykw]] num_docs_on_replica=[323]

Expected: <[DocIdSeqNoAndTerm{id='147 seqNo=149 primaryTerm=1}, DocIdSeqNoAndTerm{id='148 seqNo=150 primaryTerm=1}, DocIdSeqNoAndTerm{id='149 seqNo=151 primaryTerm=1}, DocIdSeqNoAndTerm{id='150 seqNo=152 primaryTerm=1}, DocIdSeqNoAndTerm{id='151 seqNo=153 primaryTerm=1}, DocIdSeqNoAndTerm{id='152 seqNo=154 primaryTerm=1}, DocIdSeqNoAndTerm{id='153 seqNo=155 primaryTerm=1}, DocIdSeqNoAndTerm{id='154 seqNo=156 primaryTerm=1}, DocIdSeqNoAndTerm{id='155 seqNo=157 primaryTerm=1}, DocIdSeqNoAndTerm{id='156 seqNo=158 primaryTerm=1}, DocIdSeqNoAndTerm{id='157 seqNo=159 primaryTerm=1}, DocIdSeqNoAndTerm{id='158 seqNo=160 primaryTerm=1}]>

but: was <[DocIdSeqNoAndTerm{id='147 seqNo=149 primaryTerm=2}, DocIdSeqNoAndTerm{id='148 seqNo=150 primaryTerm=2}, DocIdSeqNoAndTerm{id='149 seqNo=151 primaryTerm=2}, DocIdSeqNoAndTerm{id='150 seqNo=152 primaryTerm=2}, DocIdSeqNoAndTerm{id='151 seqNo=153 primaryTerm=2}, DocIdSeqNoAndTerm{id='152 seqNo=154 primaryTerm=2}, DocIdSeqNoAndTerm{id='153 seqNo=155 primaryTerm=2}, DocIdSeqNoAndTerm{id='154 seqNo=156 primaryTerm=2}, DocIdSeqNoAndTerm{id='155 seqNo=157 primaryTerm=2}, DocIdSeqNoAndTerm{id='156 seqNo=158 primaryTerm=2}, DocIdSeqNoAndTerm{id='157 seqNo=159 primaryTerm=2}, DocIdSeqNoAndTerm{id='158 seqNo=160 primaryTerm=2}]>
```

Reply:
+1